paddle/fluid/imperative/layer.cc

```diff
@@ -178,6 +178,11 @@ void VarBase::AddGradOps(const std::weak_ptr<OpBase>& op) {
 
 void VarBase::ClearGradient() {
   if (grad_var_) {
+    if (grad_var_->var_.IsInitialized()) {
+      PADDLE_ENFORCE_EQ(grad_var_->var_.IsType<framework::LoDTensor>(), true,
```

Comment:
Add a to do to remind change here when we support selected rows

Reply:
done